### PR TITLE
Jetpack Backup: Get storage usage info from the WPCOM API

### PR DESCRIPTION
**Pull request by mattgawarecki**
Resolves `1200412004370260-as-1200706668146543`.

#### Changes proposed in this Pull Request

* Add Rewind size to the application state tree (i.e., a reducer, actions, selectors).
* Create a query component to fetch Rewind size from the WPCOM API.
* Refactor `BackupStorageUsage` and `BackupStorageUsageUpsell` to get their data from this new part of the state tree.
* Create new hooks to encapsulate and simplify handling of Rewind size data for display purposes.
* Move storage usage "levels" into their own file so they can be shared with other components or easily modified later.

##### Known issues

* On the My Plan page, the plan icons for Backup and Security are missing. This will be addressed in a separate PR.
* The loading placeholder shown while usage data is loading, flickers a bit. This is due to a duplicate network request, and will also be addressed in a future PR.

#### Testing instructions

_**NOTE:** These instructions apply to both Calypso Blue and Calypso Green._

##### Prerequisites

* Enable the `jetpack/only-realtime-products` in your environment, either by setting its value to `true` in the environment's JSON config file, or by adding `?flags=jetpack/only-realtime-products` in your browser URL each time you visit a page to test.

##### Existing (non-tiered) products

1. Select a Jetpack site with a Backup or Security product that is not tiered (e.g., Backup Daily, Security Real-time).
2. Visit the Backup page; verify the page looks exactly the same as it does in production.
3. Visit the My Plans page; verify the page looks exactly the same as it does in production.

##### New (tiered) products

1. Select a Jetpack site and apply a subscription for a tiered Backup or Security product.
2. Visit the Backup page; after the page fully loads, verify you see a progress bar showing how much data is used vs available for your site. This information should match the values returned by `/rewind/:site/size` (bytes used) and `/rewind/:site/policies` (bytes available).
3. Visit the My Plans page; after the page fully loads, verify you see the same progress bar display below your plan name.

##### Storage usage: Backup page (non-tiered, tiered)

image

image

##### Storage usage: My Plans page (non-tiered, tiered)

image

image

### CODE REVIEW

File: client/components/backup-storage-space/usage-display.tsx [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+const UsageDisplay: React.FC< OwnProps > = ( { loading = false, usageLevel } ) => {
+	const siteId = useSelector( getSelectedSiteId ) as number;
+
+	const translate = useTranslate();
+
+	const bytesAvailable = useSelector( ( state ) => getRewindBytesAvailable( state, siteId ) );
+	const bytesUsed = useSelector( ( state ) => getRewindBytesUsed( state, siteId ) );
+	const storageUsageText = useStorageUsageText( bytesUsed, bytesAvailable );
+	const loadingText = translate( 'Calculating…', {
```
Comment (a8ci18n) [inline client/components/backup-storage-space/usage-display.tsx:31]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **23** times:
`translate( 'Calculating price' )` *ES Score: 8*

**Comment by Luchadores** [comment]
Hey @mattgawarecki 👋 Thank you for the ping. This is looking great. I only have one thing to note here. 
On the My Plan page, we're missing the Backup icon. If we don't want to display it, I would align the storage information to the left with the Free icon above. 

Here is how it looks in design:
image

**Comment by mattgawarecki** [comment]
> I only have one thing to note here. On the My Plan page, we're missing the Backup icon.

Yup, duly noted! I plan to resolve this in another PR (see "Known Issues" in my description) 👍

File: client/components/backup-storage-space/hooks.ts [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+	return useMemo( () => {
+		if ( bytesUsed === undefined ) {
+			return null;
+		}
+
+		const usedGigabytes = bytesToUnit( bytesUsed, StorageUnits.Gigabyte );
+
+		if ( bytesAvailable === undefined ) {
+			return translate( '%(usedUnitAmount).1fGB used', '%(usedUnitAmount).1fGB used', {
```
Comment (monsieur-z) [inline client/components/backup-storage-space/hooks.ts]: I don't see `usedUnitAmount` in the args. Could it be a mistake?

**Review by monsieur-z — CHANGES_REQUESTED** [review]
Code changes look sensible.

I'm seeing what's expected with a Backup tier. However, the storage indicator is not displayed for a site with a Security tier, even though the `size` endpoint is called (actually twice from the `My plan` section, in that case).

Also, since it's difficult to test the various levels of storage, it would be nice to have some units tests for files likes `client/components/backup-storage-space/hooks.ts` or `client/components/backup-storage-space/storage-usage-levels.ts`.

**Review by monsieur-z — APPROVED** [review]
Changes look good!

**Review by Luchadores — APPROVED** [review]
Giving this green light as well. 🙌

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/6581570

Thank you @mattgawarecki for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/6581570).

**Pull request merged**